source/acquaman/AMGenericStepScanController.cpp

```diff
@@ -71,88 +68,15 @@ void AMGenericStepScanController::onScanTimerUpdate()
 
 AMAction3 * AMGenericStepScanController::createInitializationActions()
 {
-	return 0;
+	return AMBeamline::bl()->createScanInitializationAction(configuration_);
 }
 
 AMAction3 * AMGenericStepScanController::createCleanupActions()
 {
-	return 0;
+	return AMBeamline::bl()->createScanCleanupAction(configuration_);
 }
 
 void AMGenericStepScanController::buildScanControllerImplementation()
 {
-	QList<AMXRFDetector *> xrfDetectors;
```

Comment:
I'm not entirely sure about this part.  This is already quite general and didn't need to be moved as it wasn't getting in anyones way for anything.  Maybe I missed something?